src/ui/Snackbar/SnackbarContext.tsx

```diff
@@ -0,0 +1,65 @@
+import { ItemType } from '@common-types';
+import { nanoid } from 'nanoid';
+import React, {
+  ComponentProps,
+  createContext,
+  ReactNode,
+  useContext,
+  useMemo,
+} from 'react';
+import { useQueue } from 'src/hooks/useQueue';
+import SnackbarList from './SnackbarList';
+
+type TSnackbar = ItemType<ComponentProps<typeof SnackbarList>['snacks']>;
+type TSnackbarContext = {
+  enqueue(alert: Pick<TSnackbar, 'severity' | 'message'>): void;
+  showError(message: ReactNode): void;
+  showSuccess(message: ReactNode): void;
+};
+const SnackbarContext = createContext<TSnackbarContext | undefined>(undefined);
```

Comment:
Use `null` instead of `undefined` as initial value due to its semanntic. `undefined` usually means uninitialized or the developer didn't bother to set its value.

Reply:
It's better this way actually.  If you check for null inside `useContext` to throw error, what if null is a valid value?

Reply:
I can't see in any scenario a value of a context should be `null`. Since context is to share value for a subtree, it will always in the format of `Record<string, any>`